src/peer_manager.rs

```diff
@@ -963,8 +959,16 @@ impl PeerManager {
     }
 
     /// Checks whether we can accept more clients.
-    pub fn can_accept_client(&self) -> bool {
-        self.client_num() < MAX_CLIENTS_PER_PROXY
+    pub fn can_accept_client(&self, ip: &IpAddr) -> bool {
```

Comment:
`IpAddr` is copyable - is it not better to pass copyable items by value instead of reference ? It's more optimised as pass by reference can prevent NRVO. Probably similar reasons why std libs also [choose this paradigm](https://doc.rust-lang.org/std/net/enum.SocketAddr.html#method.ip).